test/periphery/integration/merkle-campaign/instant/claim/claim.t.sol

```diff
@@ -1,106 +1,21 @@
-        claimInstant();
-        uint256 index1 = defaults.INDEX1();
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_StreamClaimed.selector, index1));
-        merkleInstant.claim(index1, users.recipient1, amount, merkleProof);
-    }
-
-    modifier givenNotClaimed() {
-        _;
-    }
-
-    modifier givenNotIncludedInMerkleTree() {
-        _;
-    }
-
-    function test_RevertWhen_InvalidIndex()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 invalidIndex = 1337;
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleInstant.claim(invalidIndex, users.recipient1, amount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidRecipient()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        address invalidRecipient = address(1337);
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleInstant.claim(index1, invalidRecipient, amount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidAmount()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        uint128 invalidAmount = 1337;
-        bytes32[] memory merkleProof = defaults.index1Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleInstant.claim(index1, users.recipient1, invalidAmount, merkleProof);
-    }
-
-    function test_RevertWhen_InvalidMerkleProof()
-        external
-        givenCampaignNotExpired
-        givenNotClaimed
-        givenNotIncludedInMerkleTree
-    {
-        uint256 index1 = defaults.INDEX1();
-        uint128 amount = defaults.CLAIM_AMOUNT();
-        bytes32[] memory invalidMerkleProof = defaults.index2Proof();
-        vm.expectRevert(abi.encodeWithSelector(Errors.SablierMerkleBase_InvalidProof.selector));
-        merkleInstant.claim(index1, users.recipient1, amount, invalidMerkleProof);
-    }
-
-    modifier givenIncludedInMerkleTree() {
-        _;
-    }
-
-    function test_Claim() external givenCampaignNotExpired givenNotClaimed givenIncludedInMerkleTree {
-        vm.expectEmit({ emitter: address(merkleInstant) });
+    function test_ClaimInstant() external givenCampaignNotExpired givenNotClaimed givenIncludedInMerkleTree {
```

Comment:
And change it to the following:

```solidity
function test_Claim() external override givenCampaignNotExpired givenNotClaimed givenIncludedInMerkleTree {
    vm.expectEmit({ emitter: address(merkleInstant) });
    emit Claim(defaults.INDEX1(), users.recipient1, defaults.CLAIM_AMOUNT());

    expectCallToTransfer({ to: users.recipient1, value: defaults.CLAIM_AMOUNT() });
    claim();
    assertTrue(merkleInstant.hasClaimed(defaults.INDEX1()), "not claimed");
}
```

Reply:
the check `assertTrue(merkleInstant.hasClaimed(defaults.INDEX1()), "not claimed");` is not needed because it's been done in the shared test

Reply:
changed this in: fcca68d7711eb404162e7bd96009a6be45b823fb

Reply:
Yeah but if you use the `virtual` function in the shared test, then you will have to assert it in the override functions.

Reply:
Added my suggestion in https://github.com/sablier-labs/v2-core/pull/1016.